dali/kernels/imgproc/sampler.h

```diff
@@ -199,83 +217,161 @@ struct Sampler<DALI_INTERP_LINEAR, In> {
   }
 
   template <typename T, typename BorderValue>
-  DALI_HOST_DEV T at(vec2 xy, int c) {
-    return at(xy.x, xy.y, c);
-  }
-
-  template <typename T, typename BorderValue>
-  DALI_HOST_DEV void operator()(
-      T *out_pixel,
-      float x, float y, int c,
-      BorderValue border_value) const {
-    out_pixel[c] = at<T>(x, y, c, border_value);
-  }
-
-  template <typename T, typename BorderValue>
-  DALI_HOST_DEV void operator()(
-      T *out_pixel,
-      float x, float y,
+  DALI_HOST_DEV
+  void operator()(
+      T *out_pixel, vec2 pos,
       BorderValue border_value) const {
-    Sampler<DALI_INTERP_NN, In> NN(surface);
-    x -= 0.5f;
-    y -= 0.5f;
+    Sampler2D<DALI_INTERP_NN, In> NN(surface);
+    float x = pos.x - 0.5f;
+    float y = pos.y - 0.5f;
     int x0 = floor_int(x);
     int y0 = floor_int(y);
     float qx = x - x0;
     float px = 1 - qx;
     float qy = y - y0;
 
     for (int c = 0; c < surface.channels; c++) {
-      In s00 = NN.at(x0,   y0,   c, border_value);
-      In s01 = NN.at(x0+1, y0,   c, border_value);
-      In s10 = NN.at(x0,   y0+1, c, border_value);
-      In s11 = NN.at(x0+1, y0+1, c, border_value);
+      In s00 = NN.at(ivec2(x0,   y0),   c, border_value);
```

Comment:
Can't you just call `at(vec2 pos, int c, BorderValue border_value)` ?

Reply:
No, it would needlessly call floor_int. Moreover, I need to calculate x0+1, y0+1 and it's preferable to do so in integers since in the end I need integer coordinates to fetch the pixel.

Reply:
As a side note: this code is still not optimal, since it will apply range check twice as many times as necessary. If I wanted utmost performance, these surface fetches should be done by hand and each out-of-bounds coordinate handled separately. It would also be better to have a specialized version for BorderClamp.